packages/relay-runtime/query/fetchQueryInternal.js

```diff
@@ -32,7 +32,7 @@ type RequestCacheEntry = {|
   +subscription: Subscription,
 |};
 
-const requestCachesByEnvironment = new Map();
+const requestCachesByEnvironment = new WeakMap();
```

Comment:
This should be wrapped in a `WEAKMAP_SUPPORTED` if block. 

cc  @ccemeraldeyes @psaia